server/shovel-config.ts

```diff
@@ -0,0 +1,135 @@
+import { makeConfig, toJSON } from '@indexsupply/shovel-config'
+import type { Source, Table, Integration } from '@indexsupply/shovel-config'
+import {
+  Problems,
+  Bodies,
+  Solver,
+  Dust,
+  ProblemMetadata,
+  BodyMetadata
+} from './contracts'
+import { ethers } from 'ethers'
+
+const mainnet: Source = {
+  name: 'mainnet',
+  chain_id: 1,
+  url: 'https://ethereum-rpc.publicnode.com'
+}
+
+const sepolia: Source = {
+  name: 'sepolia',
+  chain_id: 11155111,
+  url: 'https://rpc2.sepolia.org', // 'https://ethereum-sepolia-rpc.publicnode.com' rate limited
+  batch_size: 1000,
+  concurrency: 1
+}
+
+const network = sepolia.chain_id
+const contracts = Object.fromEntries(
+  [Problems, Bodies, Solver, Dust, ProblemMetadata, BodyMetadata].map(
+    (contract) => {
+      const abi = contract.abi.abi
+      return [
+        contract.abi.contractName,
+        new ethers.Contract(contract.networks[network].address, abi)
+      ]
+    }
+  )
+)
+
+const solTypeToPgType = {
+  address: 'bytea',
+  uint256: 'numeric',
+  bytes32: 'bytea',
```

Comment:
would these be better as strings? never worked with `bytea` types in a database, are they tricky?

Reply:
just tried using text type just now but it errors. I think this conversion is hardcoded in shovel (dumb that I have to specify it here 🤷)

<img width="515" alt="Screenshot 2024-05-16 at 9 04 37 AM" src="https://github.com/trifle-labs/anybody-problem/assets/282016/c2ff9d38-0c1e-4d0c-9e1f-8735e8aae9c1">

from https://indexsupply.com/shovel/docs/


It looks like it's pretty easy to query though and postgres can `encode` and `decode` strings on the way in/out:

<img width="1512" alt="Screenshot 2024-05-16 at 9 10 13 AM" src="https://github.com/trifle-labs/anybody-problem/assets/282016/c8c217d0-f933-43bd-878b-6ff52377edc6">